client/my-sites/theme/main.jsx

```diff
@@ -635,11 +635,16 @@ class ThemeSheet extends Component {
 			showTryAndCustomize,
 			isExternallyManagedTheme,
 			isWporg,
+			isLoggedIn,
 		} = this.props;
 		const placeholder = <span className="theme__sheet-placeholder">loading.....</span>;
 		const title = name || placeholder;
 		const tag = author ? translate( 'by %(author)s', { args: { author: author } } ) : placeholder;
-		const shouldRenderButton = ! retired && ! isWPForTeamsSite && ! this.shouldRenderForStaging();
+		const shouldRenderButton =
+			! retired &&
+			! isWPForTeamsSite &&
+			! this.shouldRenderForStaging() &&
+			( ! this.hasWpOrgThemeUpsellBanner() || ! isLoggedIn );
```

Comment:
Wouldn't it be semantically more accurate to use a method/function like isUserOnFreePlan in this context? The current implementation doesn't make it immediately clear that showing the button depends on whether the user is on a free plan. Instead, it gives an impression that it's based on whether we're displaying the org upsell banner or not. Could we clarify this?

Reply:
> it gives an impression that it's based on whether we're displaying the org upsell banner or not.

That's precisely the intent. The button needs to be hidden for WP.org themes if the site has a Free, Personal, or Premium plan, which are the conditions needed for the WP.org banner to be displayed.

Reply:
The proposed changes in this PR only reference the FREE plan.
> Hide the Dotorg themes activate button on free sites for logged-in users.
I can confirm that the implemented behaviour is that the button is being displayed if the user has a paid plan and is only hidden when the user has a free plan.

Reply:
> Which makes sense since if you are on a paid plan you can install the theme.

Only on the Business plan 🤔 . WP.org cannot installed if the site has a Personal or a Premium plan. If the button is visible for those plans, that's definitely unintended.

Reply:
I see, thanks for the clarification!

Reply:
Ugh, apologies for the confusion.

I'll rename the PR title, because, as you've already noticed, the Activate button is not supposed to show on Free/Personal/Premium, and only on Business/Ecommerce.

Reply:
Also FWIW, I agree with @rcrdortiz that the function is poorly named — or, it was well-named, but then it was used for counter-intuitive reasons.

I've chosen not to proceed with a rename to keep this short and to restore the original functionality (regressed in #76837).